logger.js

```diff
@@ -3,111 +3,122 @@
 const { createLogger, format, transports } = require('winston');
 
 function convertToBoolean(value) {
-    return typeof value === 'string' ? value === 'true' : value;
+  return typeof value === 'string' ? value === 'true' : value;
 }
 
 function validateHttpConfig(httpOptions) {
-    if (!httpOptions.hasOwnProperty('host')) {
-        throw new Error('The property host is omitted from config option log2httpServer. Please fill this field or remove the property.');
-    }
+  if (!httpOptions.host) {
+    console.warn('The property host is mandatory in config option log2httpServer. httpTransport won\'t be added.');
+    return false;
+  }
+  return true;
 }
 
 function generateFileTransport(level) {
```

Comment:
if you want different transports for different layer you should use level property

Reply:
that's an overkill usually all transports are open for the defined global log level